src/TableFunctions/TableFunctionRemote.cpp

```diff
@@ -110,20 +110,46 @@ void TableFunctionRemote::parseArguments(const ASTPtr & ast_function, const Cont
         }
     }
 
+    /// Cluster function may have sharding key for insert
+    if (is_cluster_function && arg_num < args.size())
+    {
+        sharding_key = args[arg_num];
+        ++arg_num;
+    }
+
     /// Username and password parameters are prohibited in cluster version of the function
     if (!is_cluster_function)
     {
         if (arg_num < args.size())
         {
-            username = get_string_literal(*args[arg_num], "Username");
+            try
+            {
+                username = get_string_literal(*args[arg_num], "Username");
+            }
+            catch (const Exception & _ [[maybe_unused]])
```

Comment:
You can write it more simple:
`catch (const Exception &)`

Reply:
Exceptions in non-exceptional case is an anti-pattern.
Let's rewrite in another way.